compiler/runtime/native_eq.c

```diff
@@ -69,17 +69,21 @@ SkipInt SKIP_native_eq_class(sk_stack_t* st, char* obj1, char* obj2) {
 
 SkipInt SKIP_native_eq_array(sk_stack_t* st, char* obj1, char* obj2) {
   SKIP_gc_type_t* ty1 = get_gc_type(obj1);
-  SKIP_gc_type_t* ty2 = get_gc_type(obj2);
+  {
+    SKIP_gc_type_t* ty2 = get_gc_type(obj2);
 
-  if (ty1 != ty2) {
-    return 1;
+    if (ty1 != ty2) {
+      return 1;
+    }
   }
 
   size_t len1 = skipArrayLen(obj1);
-  size_t len2 = skipArrayLen(obj2);
+  {
+    size_t len2 = skipArrayLen(obj2);
 
-  if (len1 != len2) {
-    return 1;
+    if (len1 != len2) {
+      return 1;
+    }
```

Comment:
Not objecting, but why do this?

Reply:
Right, using `len2` instead of `len1` wouldn't be an issue... I'll revert